mlflow/utils/proto_json_utils.py

```diff
@@ -99,14 +99,25 @@ def _dataframe_from_json(
     from mlflow.types import DataType
 
     if schema is not None:
-        dtypes = dict(zip(schema.input_names(), schema.pandas_types()))
+        if schema.is_tensor_spec():
+            # The schema can be either:
+            #  - a single tensor: attempt to parse all columns with the same dtype
+            #  - a dictionary of tensors: each column gets the type from ____ named tensor
```

Comment:
Sorry I kept it as `____` previously if a better name came to mind.
```suggestion
            #  - a dictionary of tensors: each column gets the type from an equally named tensor
```